vpc.tf

```diff
@@ -18,7 +18,7 @@ resource "aws_route_table" "route-table" {
   vpc_id = aws_vpc.core-infra.id
   
   route {
-    cidr_block = "10.0.0.0/24"
+    cidr_block = "10.0.0.0/16"
     gateway_id = aws_internet_gateway.gw.id
   }
  }
```

Comment:
You need to associate the route table to the subnet. Follow this guide https://www.terraform.io/docs/providers/aws/r/route_table_association.html